testdata/src/go.uber.org/anonymousfunction/anonymous_functions_assignment.go

```diff
@@ -18,15 +18,15 @@ package anonymousfunction
 
 func testSimpleAssignment() {
 	var t1 *int
-	a := func() { // want "value is read from a variable that was never assigned to"
-		print(*t1)
+	a := func() {
+		print(*t1) //want "read from a variable that was never assigned to"
```

Comment:
Interesting. What is causing us to switch these to get reported here vs at lambda declaration? (This is better, but I wonder if it's an intentional change or some consequence of the new error reporting method)

Edit: Just realized that the code above keeps updating the conflict position for as long as it finds nodes in its non-nil path, so the reporting position of the conflict ends up being that of the final non-nil consumer in the flow, which is what we want. Nice! (Maybe worth a once sentence comment in the code where this happens)

Reply:
Exactly, you got the reasoning right. Added a comment explaining this idea.